pallets/sgx-account-linker/src/tests.rs

```diff
@@ -0,0 +1,436 @@
+
+		let index = 0_u32;
+		let network_type = crate::NetworkType::Kusama;
+
+		let bytes = generate_sub_raw_message(&account.clone(), network_type, block_number);
+		let msg = sp_io::hashing::keccak_256(&bytes);
+		let signature_raw = generate_sr25519_sig(msg);
+
+		//signature is 4cdfda29585e33fe1969ee67bab9d29e440973b8cfde47346043c91d6d763c317659686089c2642e9a64fa28a00b1767fd134484fc0146de6d9eefdf366d2c81
+		let signature = crate::MultiSignature::Sr25519Signature(signature_raw.0);
+
+		assert_ok!(SgxAccountLinker::do_link_sub(
+			account.clone(),
+			index,
+			network_type,
+			account.clone(),
+			block_number,
+			layer_one_blocknumber,
+			signature
+		));
+	});
+}
+
+#[test]
+fn test_link_sub_ed25519_address() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+
+		// account id of Alice 0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee
+		let account: AccountId32 = AccountId32::from([
+			136, 220, 52, 23, 213, 5, 142, 196, 180, 80, 62, 12, 18, 234, 26, 10, 137, 190, 32, 15,
+			233, 137, 34, 66, 61, 67, 52, 1, 79, 166, 176, 238,
+		]);
+
+		let block_number: u32 = 10000;
+		let layer_one_blocknumber: u32 = 10;
+
+		let index = 0_u32;
+		let network_type = crate::NetworkType::Kusama;
+
+		let bytes = generate_sub_raw_message(&account.clone(), network_type, block_number);
+		let msg = sp_io::hashing::keccak_256(&bytes);
+
+		// signature is 0x08b1284cdaf008c80740d52be923cf24d45f8ba6e009bfd61a0a364c23df98c268dff3a8baabb47c8011ba39f76729aeab3c0281bfa45ef9389162fd78c18b08
+		let signature_raw = generate_ed25519_sig(msg);
+
+		let signature = crate::MultiSignature::Ed25519Signature(signature_raw.0);
+
+		assert_ok!(SgxAccountLinker::do_link_sub(
+			account.clone(),
+			index,
+			network_type,
+			account.clone(),
+			block_number,
+			layer_one_blocknumber,
+			signature
+		));
+	});
+}
+
+#[test]
+fn test_link_sub_ecdsa_address() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+
+		// account id of Alice 0x01e552298e47454041ea31273b4b630c64c104e4514aa3643490b8aaca9cf8ed
+		let account: AccountId32 = AccountId32::from([
+			1, 229, 82, 41, 142, 71, 69, 64, 65, 234, 49, 39, 59, 75, 99, 12, 100, 193, 4, 228, 81,
+			74, 163, 100, 52, 144, 184, 170, 202, 156, 248, 237,
+		]);
+
+		let block_number: u32 = 10000;
+		let layer_one_blocknumber: u32 = 10;
+
+		let index = 0_u32;
+		let network_type = crate::NetworkType::KusamaParachain(1);
+
+		let bytes = generate_sub_raw_message(&account.clone(), network_type, block_number);
+
+		// signature is 0xbf9484a706e5fadbd9ae8fd2e61f58c8aea387816903ef5b549af19cbf8c4fd831782058ea8f7acddc0e024f4d1ca155052fb04ad4115eeed75fefd6a7a6764301
+		let signature_raw = generate_ecdsa_sig(&bytes[..]);
+
+		let signature = crate::MultiSignature::EcdsaSignature(signature_raw.0);
+
+		assert_ok!(SgxAccountLinker::do_link_sub(
+			account.clone(),
+			index,
+			network_type,
+			account.clone(),
+			block_number,
+			layer_one_blocknumber,
+			signature
+		));
+	});
```

Comment:
You're only testing the happy flow of sub links. Might be worth to test unhappy as well.